ImguiMapEditor/Controllers/SearchController.cpp

```
@@ -75,8 +77,94 @@ void SearchController::onMapLoaded(
+    if (!map_search_service_) return;
+    async_search_future_ = std::async(std::launch::async,
+        [service = map_search_service_.get()]() {
+            return service->searchByWriteable();
+        });
+    async_search_active_ = true;
+}
+
+void SearchController::searchTextAsync(const std::string& query, bool search_items, bool search_creatures) {
+    if (!map_search_service_ || query.empty()) return;
+
+    bool is_number = !query.empty() && std::all_of(query.begin(), query.end(), ::isdigit);
```

Clang (14.0.6)

[warning] variable 'is_number' is not initialized

(cppcoreguidelines-init-variables)